### PR TITLE
add support for split model file url

### PR DESCRIPTION
**Pull request by fozzee**
fix local model file url returning early before assembling split model file list

## Summary by Sourcery

Add support for split model file URLs by assembling and handling individual split files for both local and remote models, fix premature return when processing local model URLs, and improve file copying behavior.

New Features:
- Add support in pull() for split model file URLs by generating a list of individual split files

Bug Fixes:
- Fix premature return for local model file URLs that prevented assembling split model file list

Enhancements:
- Use LocalModelFile with generated SHA256 hashes for local split model files

## Summary by Sourcery

Support split model file URLs by properly assembling split files for both local and remote models, fix local URL handling order, and improve hash generation for split files

New Features:
- Support pulling split model file URLs by assembling individual split segments

Bug Fixes:
- Fix premature return that prevented assembling split model files for local file URLs

Enhancements:
- Use LocalModelFile with generated SHA256 hashes for local split model files and apply SHA256 hashing for remote segments

### CODE REVIEW

**Comment by rhatdan** [comment]
@olliewalsh @engelmi PTAL

File: ramalama/transports/url.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def download(self, blob_file_path, snapshot_dir):
         if not os.path.exists(self.url):
             raise FileNotFoundError(f"No such file: '{self.url}'")
         # moving from the local location to blob directory so the model store "owns" the data
-        shutil.copy(self.url, blob_file_path)
+        shutil.copy(self.url, blob_file_path, follow_symlinks=False)
```
Comment (engelmi) [inline ramalama/transports/url.py]: If I understand it correctly, this means we only copying the symlink. So on the one side we don't use more disk space and on the other we can't guarantee that the file the symlink is pointing to gets removed. I'd prefer copying the actual file to be in control and avoid split situations.

**Comment by olliewalsh** [comment]
This used to work (https://github.com/containers/ramalama/pull/1426), has there been a regression?

File: ramalama/transports/url.py (hunk 1 above)
Comment (fozzee) [reply ramalama/transports/url.py]: Yes, this will copy the model file without following or resolving symlinks. This only changes the current behavior of ramalama if the user uses a symlink as the model url.

I agree it isn't ideal. However, I also think it is unexpected behavior from the user perspective to follow a symlink and make a copy of the target model file if the user is providing a symlink to the model file.

I will remove this for now since this implementation is naive and breaks if the symlink is targeting a relative path. This PR can just focus on fixing the split model file:// url.

**Comment by fozzee** [comment]
> This used to work (https://github.com/containers/ramalama/pull/1426), has there been a regression?

Yes. It looks like the split model logic was refactored into the URL tranport and it didn't account for a code path including both split model files and the file:// url usage. Also, the url split model unit test only tests https:// urls.

There seem to be some other corner case issues like using an absolute path instead of a relative path. And running `ramalama ls` shows a model url for file:// urls that assumes it is an absolute path with an extra prefixed `/`. So there is probably some further review of this area of the code needed.

File: ramalama/transports/url.py (hunk 1 above)
Comment (fozzee) [reply ramalama/transports/url.py]: This is removed now.

**Comment by fozzee** [comment]
Removed the `follow_symlinks=False` from the commit and removed `Fixes: #1989` from prior comment to continue discussion about how to handle use of models with model files external to model store.

I will leave this PR as just the fix for split model file urls.

File: ramalama/transports/url.py [hunk 2]
```diff
@@ -92,17 +92,28 @@ def _assemble_split_file_list(self, snapshot_hash: str) -> list[SnapshotFile]:
         for i in range(1, total_parts + 1):
```
[the remark this answers is not in the document]
Comment (engelmi) [reply ramalama/transports/url.py:92]: Not the goal of this PR to refactor this part. Could be done in a separate one.
